pyproject.toml

```diff
@@ -104,13 +105,32 @@ version-file = "anndata/_version.py"
 
 [tool.coverage.run]
 source = ["anndata"]
-omit = ["setup.py", "versioneer.py", "anndata/_version.py", "**/test_*.py"]
+omit = ["anndata/_version.py", "**/test_*.py"]
 
 [tool.pytest.ini_options]
-addopts = "--doctest-modules"
+addopts = [
+    "--strict-markers",
+    "--doctest-modules",
+    "--ignore=anndata/core.py",      # deprecated
+    "--ignore=anndata/readwrite.py", # deprecated
+]
 python_files = "test_*.py"
 testpaths = ["anndata", "docs/concatenation.rst"]
-filterwarnings = ['ignore:X\.dtype being converted to np.float32:FutureWarning']
+filterwarnings = [
+    'error',
+    # We still test deprecated functions
+    'ignore:The AnnData\.concatenate method is deprecated:FutureWarning',
+    'ignore:`anndata\.read` is deprecated:FutureWarning',
+    # We test our experimental features too
+    'ignore::anndata._warnings.ExperimentalFeatureWarning',
+    # Third party issues
+    'ignore::numba.core.errors.NumbaDeprecationWarning', # we don’t import numba
+    # TODO: Catch the following ones in tests or fix them
+    'default::anndata._warnings.ImplicitModificationWarning',
+    'default:.*names are not unique. To make them unique:UserWarning',
+    'default::scipy.sparse._base.SparseEfficiencyWarning',
+    'default::dask.array.core.PerformanceWarning',
```

Comment:
- our `ImplicitModificationWarning` and `names are not unique` warnings should be fixed in lib code and caught in test code so we can be sure none of our functions/methods invoke them, only user/test code.
- scipy `SparseEfficiencyWarning` and dask `PerformanceWarning` should be fixed, they probably indicate real performance issues